libs/contrib/Data/Void.idr

```diff
@@ -0,0 +1,13 @@
+module Data.Void
+
+export
+persistentVoid : (0 x : Void) -> Void
+persistentVoid _ impossible
+
+export
+absurdity : Uninhabited t => (0 _ : t) -> s
+absurdity x = absurd (persistentVoid (uninhabited x))
+
+export
+absurdSince : (Uninhabited t) => (0 _ : x -> t) -> (x -> s)
+absurdSince since x = absurdity (since x)
```

Comment:
I would probably call this `contradiction` as it's essentially
`(¬ x /\ x) -> y`. Note that `x` could be made `0`.

Reply:
Renamed.

> Note that `x` could be made `0`.

Noted, but since `Not` is defined to be `a -> Void`, we can't use `contradiction` in place of a `Not a`.

We could change the library so that `Not a` takes an irrelevant `a`.

Reply:
I'm in favour of keeping it as is `contradiction (Uninhabited t) => (0 _ : x -> t) -> (x -> s)`  for the same reason we keep `absurd` as is.